cmd/zpool/zpool_main.c

```diff
@@ -3963,11 +4069,89 @@ fsleep(float sec)
 	nanosleep(&req, NULL);
 }
 
+/* Print a "zpool status/iostat -c" script's help (-h) output. */
+static void
+print_zpool_script_help(char *name, char *path)
+{
+	FILE *fp;
+	char *line = NULL;
+	int link[2];
+	pid_t pid;
+	int status;
+	size_t len = 0;
+	char *argv[] = {path, "-h", NULL};
+	char *nl = NULL;
+
+	if (pipe(link) == -1)
+		return;
+
+	if ((pid = fork()) == -1)
+		return; /* Couldn't fork */
+
+	if (pid == 0) {
+		/* Child process */
+		dup2(link[1], STDOUT_FILENO);
+		close(link[0]);
+		close(link[1]);
+		execv(path, argv);
+	} else {
+		wait(&status);
+		close(link[1]);
+		fp = fdopen(link[0], "r");
+		if (fp == NULL)
+			return;
+
+		/* Read back result from child process */
+		if (getline(&line, &len, fp) != -1) {
+			/* Remove newline */
+			if ((nl = strchr(line, '\n')) != NULL)
+				*nl = '\0';
+
+			if (!is_blank_str(line))
+				printf("  %-14s  %s\n", name, line);
+		}
+
+		free(line);
+		fclose(fp);
+	}
+}
 
 /*
- * zpool iostat [-c CMD] [-ghHLpPvy] [[-lq]|[-r|-w]] [-n name] [-T d|u]
- *		[[ pool ...]|[pool vdev ...]|[vdev ...]]
- *		[interval [count]]
+ * Print a list of "zpool status/iostat -c" scripts
+ */
+static void
+print_zpool_script_list(void)
+{
+	DIR *dir;
+	struct dirent *ent;
+	char fullpath[MAXPATHLEN];
+	struct stat dir_stat;
+	if ((dir = opendir(ZPOOL_SCRIPTS_DIR)) != NULL) {
+		printf("\n");
+		/* print all the files and directories within directory */
+		while ((ent = readdir(dir)) != NULL) {
+			sprintf(fullpath, "%s/%s", ZPOOL_SCRIPTS_DIR,
+			    ent->d_name);
+
+			/* Print the scripts */
+			if (stat(fullpath, &dir_stat) == 0)
+				if (dir_stat.st_mode & S_IXUSR &&
+				    S_ISREG(dir_stat.st_mode))
+					print_zpool_script_help(ent->d_name,
+					    fullpath);
+		}
+		printf("\n");
+		closedir(dir);
+	} else {
+		fprintf(stderr, gettext("ERROR: Can't open %s scripts dir\n"),
```

Comment:
No need for the `ERROR:` here which differs from all the other error messages.   Please remove it.